tests/lms/config/settings_test.py

```diff
@@ -2,53 +2,120 @@
 
 import pytest
 
-from lms.config import settings
+from lms.config.settings import SettingError, SettingGetter
 
 
 @pytest.mark.usefixtures("os_fixture")
-class TestEnvSetting:
-    def test_it_returns_the_value_from_the_environment_variable(self, os_fixture):
-        os_fixture.environ = {"FOOBAR": "the_value"}
-
-        result = settings.env_setting("FOOBAR")
-
-        assert result == "the_value"
-
-    def test_it_returns_none_when_environment_variable_isnt_set_and_optional(
-        self, os_fixture
-    ):
-        os_fixture.environ = {}
-
-        result = settings.env_setting("FOOBAR")
-
-        assert result is None
-
-    def test_it_raises_if_the_environment_variable_isnt_set_and_required(
-        self, os_fixture
+class TestSettingGetter:
+    @pytest.mark.parametrize(
+        "environ,settings,envvar_name,required,default,expected",
+        [
+            ({"FOO_BAR": "from_environ"}, {}, "FOO_BAR", False, None, "from_environ"),
```

Comment:
When there are 5 positional parameters and one output, it is difficult to see at a glance what each case is testing. A comment above each would help. Also, if there are any parameters which are not varied between tests, omitting those helps as well.